packages/@aws-cdk/aws-vpclattice-alpha/lib/service-network-base.ts

```diff
@@ -0,0 +1,61 @@
+import { CfnServiceNetworkServiceAssociation } from 'aws-cdk-lib/aws-vpclattice';
+import { IService } from './service';
+import { IResource, Resource } from 'aws-cdk-lib/core';
+
+/**
+ * Interface representing a VPC Lattice service network
+ */
+export interface IServiceNetwork extends IResource {
+  /**
+   * The ARN of the service network
+   * @attribute
+   * @returns a token representing the service network ARN
+   */
+  readonly serviceNetworkArn: string;
+
+  /**
+   * The ID of the service network
+   * @attribute
+   * @returns a token representing the service network ID
+   */
+  readonly serviceNetworkId: string;
+
+  /**
+   * Associate services with this service network
+   *
+   * @param services The services to associate with this service network
+   */
+  associateServices(services: IService[]): void;
+}
+
+/**
+ * Base class for VPC Lattice service networks
+ */
+export abstract class ServiceNetworkBase extends Resource implements IServiceNetwork {
```

Comment:
Guidelines say not to export the base class, but there's no way around it if the base class is in a different file (which is not necessary, of course).

Is it a hard requirement not to expose the base class?

Reply:
The important part will be that it's not exported from `index.ts`

Reply:
I tried that - without the `index.ts` export, the import in `service-network.ts` does not work (or I am doing something wrong).

DDB Tablev2 also exports the base class FYI: https://github.com/aws/aws-cdk/blob/84e875a54b2a32a4edb07c870139415d0a73b6b9/packages/aws-cdk-lib/aws-dynamodb/lib/index.ts#L5 (not saying this is correct, just pointing it out).